Firestore/core/src/firebase/firestore/api/write_batch.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_WRITE_BATCH_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_WRITE_BATCH_H_
+
+#if !defined(__OBJC__)
+#error "This header only supports Objective-C++"
+#endif  // !defined(__OBJC__)
+
+#import <Foundation/Foundation.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/core/user_data.h"
```

Comment:
FWIW- I could avoid this include with forward declarations, but I'd have to do something like this below:

```
namespace firebase {
namespace firestore {
namespace core {
  class ParsedSetData;
  class ParsedUpdateData;
}
namespace api {
...
```

which I assume isn't worth it? (I don't see us doing that elsewhere...)  In general it doesn't seem like we're particularly dedicated to using forward declarations, so maybe I shouldn't be bothering at all...  Not sure.

Reply:
The style guide [recommends](https://google.github.io/styleguide/cppguide.html#Forward_Declarations) to always prefer includes to forward declarations (unless impossible, e.g. in case of a cyclical reference). I want to experiment with mass-changing our current style-guide-compliant approach to the more common C++ way of preferring forward declarations, but Gil has previously pointed out that a bunch of our compilation times comes from Abseil and gRPC, which wouldn't be affected.

Reply:
Oh! Interesting.  I don't think we follow that universally, but I went ahead and switched to full includes here.

Reply:
Incidentally, I'm favor of the experiment and I believe it could ultimately be useful even if it doesn't benefit end users that much because it could reduce our incremental compile times.

I'm also in favor of having clear before and after numbers to start from, so starting with `#include` universally seems better than muddying things initially.